### DISCUSSION TITLE
"Recording remuxed, but the file may be incomplete" PCM 32 Bit Float audio codec

### DISCUSSION THREAD

**Discussion by IssacI19** [obsproject/obs-studio#9174] [Development Questions] [answered]
I've been testing out new settings for recording in OBS, and I want to use the "FFmpeg PCM (32-bit float)" audio codec in the recording, but I'm having this weird thing where when I remux the .mkv file to .mp4 in obs, I get a popup that says "Recording remuxed, but the file may be incomplete". It's interesting because the original recording, the .mkv file, is totally fine. It's just when I remux it to .mp4 with that same codec. This goes for any of the other FFmpeg PCM audio codecs. The file is multi-track audio, just wanted to say. Someone, please help

![Screenshot 2023-06-29 075718](https://github.com/obsproject/obs-studio/assets/137911543/7071a281-a6b0-40cd-9f64-ddf745a01ef2)
![Screenshot 2023-06-29 075804](https://github.com/obsproject/obs-studio/assets/137911543/904e73b7-3377-4bc2-a5bc-4963b0727e1b)
![Screenshot 2023-06-29 075854](https://github.com/obsproject/obs-studio/assets/137911543/8475d663-d317-4f54-91c2-dfecc046c800)
![Screenshot 2023-06-29 080118](https://github.com/obsproject/obs-studio/assets/137911543/fd663f94-ebc5-40bb-9ebf-01db10a65596)

**Comment by tytan652** [comment, marked as answer]
FFmpeg 6.0 does not support PCM inside of a MP4 container.

**Reply by IssacI19** [reply]
how do you know what version of FFmpeg you have

**Reply by IssacI19** [reply]
So I would want to use MOV or MKV instead for PCM audio?

**Reply by tytan652** [reply]
> how do you know what version of FFmpeg you have

Windows and macOS builds of OBS have FFmpeg 6.0.

> So I would want to use MOV or MKV instead for PCM audio?

You are already using MKV before remuxing.

**Reply by IssacI19** [reply]
So what would happen if I use "Fragmented MOV", do you still have to remux a "Fragmented MOV" file?

**Reply by IssacI19** [reply]
Because you were right I remixed it to MOV and it works.l but now when I bring it into Premiere or it shows up with one audio track when the file has 3 tracks. It shows up in Media Player Classic, but in Premiere only 1

**Reply by ElectronicWar** [reply]
Worst case is to use mkvtoolnix to demux the MKV into it's elementary video and audio stream and import them separately into Premier if needed.

**Reply by IssacI19** [reply]
I have downloaded Davinci Resolve 28 to see if its the file or just Premiere Pro and it pops up as multi-channel audio

**Comment by ElectronicWar** [comment]
PCM audio is currently only supported in MOV and MKV containers in ffmpeg (and in extension OBS).
AV1 video is in turn only supported in MP4 and MKV containers.

So for your usecase MKV is the only valid container at the moment.
